discordbot/commands/drc.py

```diff
@@ -82,29 +96,32 @@ async def matchups(ctx: MtgContext, *, args: str) -> None:
         domain=link_domain, length=output['length'], wr=output['winrate'], url=output['compress'])
     await ctx.send(ans)
 
-def search_dreadrise(query: str) -> Tuple[int, List[str]]:
+async def search_dreadrise(query: str) -> Union[Tuple[str, None], Tuple[int, List[str]]]:
     """Returns a tuple. First member is an integer indicating how many cards match the query total,
        second member is a list of card names up to the maximum that could be fetched in a timely fashion."""
```

Comment:
I think rather than making this convoluted Union type the right answer here is to return three values.

    how_many, cards, error = await search_dreadrise(query)
    if error:
        # handle the error
        return
    # proceed to handle the non-error case
    
Then your type is just:

    Tuple[int, List[str], str]

or maybe:

    Tuple[int, List[str], Optional[str]]
    
if you want to return `None` and not the empty string for "no error". I think it's fine/good to return a `how_many` of 0 and and an empty list of cards in the error case.

Reply:
i like this approach, will do